core/store/src/flat/store_helper.rs

```diff
@@ -174,3 +174,73 @@ pub fn key_belongs_to_shard(
     let (key_shard_uid, _) = decode_flat_state_db_key(key)?;
     Ok(key_shard_uid.version == shard_layout.version() && key_shard_uid.shard_id as u64 == shard_id)
 }
+
+#[cfg(test)]
+mod tests {
+    use super::iter_flat_state_entries;
+    use crate::flat::store_helper::{set_delta, set_flat_state_value};
+    use crate::flat::{
+        BlockInfo, FlatStateChanges, FlatStateDelta, FlatStateDeltaMetadata, FlatStateValue,
+        FlatStorageManager,
+    };
+    use crate::test_utils::create_test_store;
+    use crate::CryptoHash;
+    use near_primitives::shard_layout::ShardUId;
+
+    #[test]
+    fn test_iter_flat_state_entries() {
+        // Setup shards and store
+        let store = create_test_store();
+        let shard_uids = Vec::from([
```

Comment:
nit: `[0, 1, 2].map(|id| ShardUId { shard_id: id, version: 0 })` to reduce code duplication